ksql-engine/src/main/java/io/confluent/ksql/function/udf/array/Range.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright 2019 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"; you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.function.udf.array;
+
+import io.confluent.ksql.function.udf.Udf;
+import io.confluent.ksql.function.udf.UdfDescription;
+import io.confluent.ksql.function.udf.UdfParameter;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+import java.util.stream.LongStream;
+
+/**
+ * This UDF constructs an array containing an array of INTs or BIGINTs in the specified range
+ */
+@UdfDescription(name = "RANGE", description = "Construct an array of a range of values")
```

Comment:
if we're following postgres on this one, we would name it `generate_series`  and it would have an optional `step` parameter. cc @hjafarpour @blueedgenick for their opinions on how sql-standardy we should be (RANGE might have another meaning in standard SQL)